tests/jira/OnPremOAuth.ts

```diff
@@ -0,0 +1,247 @@
+            expect(usesCustomPort(new URL('http://example.com:8008'))).to.be.true;
+            expect(usesCustomPort(new URL('http://example.com:8080'))).to.be.true;
+        });
+        it("returns true for a non-default HTTPS port", () => {
+            expect(usesCustomPort(new URL('https://example.com:1234'))).to.be.true;
+            expect(usesCustomPort(new URL('https://example.com:8443'))).to.be.true;
+        });
+        it("returns false when no port is specified", () => {
+            expect(usesCustomPort(new URL('http://example.com'))).to.be.false;
+        });
+        it("returns false for default ports", () => {
+            expect(usesCustomPort(new URL('http://example.com:80'))).to.be.false;
+            expect(usesCustomPort(new URL('https://example.com:443'))).to.be.false;
+        });
+    });
+});
+
+describe("JiraOnPremOAuth helper functions", () => {
+    const config: BridgeConfigJiraOnPremOAuth = {
+        consumerKey: "abc",
+        privateKey: "abc.pem",
+        // eslint-disable-next-line camelcase
+        redirect_uri: "https://redirect",
+    };
+    // This private key is ONLY used for testing. It's safe to be published.
+    const privateKey = `-----BEGIN PRIVATE KEY-----
+MIIBVgIBADANBgkqhkiG9w0BAQEFAASCAUAwggE8AgEAAkEA0HOkgIfQZHftA0+c
+zj0S5/jWvacDOpA6Fe3AglOuGzqpB0R/2LjF7XroNm1SsoFqhOTcq/rYmBN7k11C
+04aXtQIDAQABAkEAl1Yd7CKuLQLUVD8MfL7iZv/GB9KlCXOEeD8wMG5ITQMt8CVQ
+LbiikP/+NhQPEt7nIaCVkeGHSWUwRPQQ5c0e0QIhAO7HVRDlIOrWFQvclq2HeaHC
+4YmHG5hCFJ+yoahBvA5fAiEA33xfQhWHSFkW93UCEdYucqWbHwxionr9G0/pb1gP
+KmsCIQCchH8nTvpv1RsApn0CjK5XMZaDftPAz3gTYpWC5GvwAwIgH/WNOcoSWopC
++ohFD3/tjH+aO0puIIYuA+XdoAqbwJkCIQDZM/4Q0W9VWi6irF/SuImhWPzOIvfU
+0XPlj5/qcwcSyA==
+-----END PRIVATE KEY-----
+`;
+
+    // Start: HTTP mock server
+    let mockServer: Server;
+    const mockServerPort = 19473;
+    const mockServerAddress = `http://localhost:${mockServerPort}`;
+    // Tests may change this, to chnage the HTTP response body.
+    let mockServerResponse: Record<string, string>;
+    before(async () => {
+        mockServer = createServer((_req, res) => {
+            res.end(JSON.stringify(mockServerResponse));
+        });
+        await new Promise<void>((resolve, reject) => {
+            mockServer.listen(mockServerPort, resolve);
+            mockServer.on('error', reject);
+        });
+    });
+    after(async () => {
+        const closeServer = promisify(mockServer.close.bind(mockServer));
+        await closeServer();
+    });
+    // End: HTTP mock server
+    describe("exchangeRequestForToken", () => {
+        it("fetches oauth tokens to return access_token", async() => {
+            const jiraOAuth = new JiraOnPremOAuth(config, mockServerAddress, Buffer.from(privateKey));
+            mockServerResponse = {
+                oauth_token: "abc",
+                oauth_token_secret: "def",
+            };
+            const actual = await jiraOAuth.exchangeRequestForToken("foo", "bar");
```

Comment:
If mocha runs the tests in parallel, this is a race condition with the other tests.